src/hooks/usePagination.js

```diff
@@ -1,14 +1,30 @@
 export const DOTS = "...";
 
-function usePagination() {
+function usePagination({ currentPage, totalCount, pageSize }) {
```

Comment:
The value returned by this hook can be memoized.cached to avoid unnecessarily running the logic every time the hook is called. The useMemo() seems suitable.